### PR TITLE
feat: allow conference participant to be any endpoint

### PR DESCRIPTION
**Pull request by JaymoKang**
# Fixes #

Allows clients and sip to also be conference participants
First time contributing in this repository, I have no idea what I'm doing. Any help is appreciated :)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the master branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/rest/api/v2010/account/conference/Participant.java [hunk 1]
```diff
@@ -140,8 +140,8 @@ public static ParticipantUpdater updater(final String pathConferenceSid,
      */
     public static ParticipantCreator creator(final String pathAccountSid,
                                              final String pathConferenceSid,
-                                             final com.twilio.type.PhoneNumber from,
-                                             final com.twilio.type.PhoneNumber to) {
+                                             final com.twilio.type.Endpoint from,
```
Comment (eshanholtz) [inline src/main/java/com/twilio/rest/api/v2010/account/conference/Participant.java:143]: I'm not sure I understand the purpose/use case for this change. These types are autogenerated from our internal API definition, and the type expected here is a `com.twilio.type.PhoneNumber`.

Comment (eshanholtz) [reply src/main/java/com/twilio/rest/api/v2010/account/conference/Participant.java:143]: Ah I found the docs mentioning the alternate `to`/`from` formats [here](https://www.twilio.com/docs/voice/api/conference-participant-resource#create-a-participant)

**Comment by eshanholtz** [comment]
This PR LGTM but requires downstream API changes to autogenerate properly. An internal ticket has been filed against the relevant product team (reference ID: VOICE-6421). I'm going to hold off approving and merging until the upstream changes have been made.

**Comment by eshanholtz** [comment]
@JaymoKang please check the license agreement in the checklist.

**Comment by JaymoKang** [comment]
@eshanholtz any updates?

**Comment by childish-sambino** [comment]
Still waiting on the internal ticket.

**Comment by thinkingserious** [comment]
Internal reference ID update: CONF-2813

File: src/main/java/com/twilio/rest/api/v2010/account/conference/Participant.java [hunk 2]
```diff
@@ -489,4 +489,24 @@ public int hashCode() {
...
+                          .add("dateUpdated", dateUpdated)
+                          .add("endConferenceOnExit", endConferenceOnExit)
+                          .add("muted", muted)
+                          .add("hold", hold)
+                          .add("startConferenceOnEnter", startConferenceOnEnter)
+                          .add("status", status)
+                          .add("uri", uri)
+                          .toString();
+    }
+}
```
Comment (JaymoKang) [inline src/main/java/com/twilio/rest/api/v2010/account/conference/Participant.java:512]:
```suggestion
```

File: src/main/java/com/twilio/rest/api/v2010/account/conference/Participant.java [hunk 3]
```diff
@@ -489,4 +489,24 @@ public int hashCode() {
...
+                          .add("dateCreated", dateCreated)
+                          .add("dateUpdated", dateUpdated)
+                          .add("endConferenceOnExit", endConferenceOnExit)
+                          .add("muted", muted)
+                          .add("hold", hold)
+                          .add("startConferenceOnEnter", startConferenceOnEnter)
+                          .add("status", status)
+                          .add("uri", uri)
+                          .toString();
+    }
```
Comment (JaymoKang) [inline src/main/java/com/twilio/rest/api/v2010/account/conference/Participant.java:511]:
```suggestion
```

**Review by eshanholtz — APPROVED** [review]
We received confirmation that this has been fixed on the back end, so I'm going to go ahead and merge this PR. Thank you @JaymoKang for the contribution and your patience getting this merged!

**Pull request merged**